### DISCUSSION TITLE
Installing iOS apps on M1 devices?

### DISCUSSION THREAD

**Discussion by cristim** [Homebrew/discussions#1510] [Tap maintenance and Homebrew development] [answered]
Would it be possible to add support for installing iOS apps on M1 hardware, much like installing casks?

**Comment by carlocab** [comment, marked as answer]
If they're distributed like casks (e.g. `pkg` installers, or `dmg` images), then probably. If not, it'll probably require changes to `brew`.

If you can install them from the Mac App Store, then you might be able to use `mas`. (`brew install mas`)

**Reply by cristim** [reply]
Thanks for the quick reply, I'll check out `mas`.

This was just an idea for future implementation, I'm aware it would require changes to brew and adding some additional type of installation assets besides casks, I was just wondering if this is doable and how hard would it be to do it.

Besides the ease of installation from a few keystrokes, I was thinking that it might be useful to have direct support into brew for portability reasons, such as `brew bundle (dump)`

**Reply by carlocab** [reply]
`brew bundle` also records things you install via `mas`. If you can install them through `mas`, then that should be sufficient, no? We don't want to remake the wheel here.

**Reply by cristim** [reply]
yes, that should be fine. `mas` seems pretty nice, already figured it out to install stuff and if we have the `brew bundle` integration we should be good.

I just find a bit unfriendly its use of numeric IDs instead of human-friendly names like we have with casks, but probably there's no other way and anyway that's another story :-)

**Reply by cristim** [reply]
Just got another idea speaking of this numeric ID thing I complained about above, could `brew` maybe use `mas` as a sort of provider and maintain human-friendlier names for `mas` apps, essentially allowing installation of apps using `mas` under the hood using `brew` commands, something like

`brew install garageband` instead of `mas install 682658836`?

**Reply by carlocab** [reply]
Surely that needs to be fixed in `mas`? I mean, if `mas` can't do it, I don't see how it's possible for `brew` to. `brew` doesn't even know how to query the App Store.

**Reply by cristim** [reply]
Yeah, you're probably right, never mind this.

**Reply by SMillerDev** [reply]
App names are not guaranteed to be unique AFAIK. So `mas install x` is not possible because that isn't enough info.

**Reply by cristim** [reply]
Yes, I was only referring to the IDs and having brew store a mapping of brew-internal names  corresponding  to the mas IDs more or less similar to the mapping between the arbitrary brew-internal cask names and the DMG URLs

**Reply by SMillerDev** [reply]
So... Like recreating the app store on top of the app store?

**Reply by cristim** [reply]
no, just aliasing the ID to a cleaner name more consistent to the other brew package names (no special characters, spaces, etc.) and ensured to be unique within brew since the Appstore names are free-form and not guaranteed to be unique.

Probably the brew package names could be generated automatically from the AppStore package names, although that might be hard, otherwise this would need to be manually maintained (I can see how this might be hard over long term as well with hundreds of thousands of apps in the Appstore).

Under the hood it would still call `mas` with the same IDs, etc.

But yeah, this was just a thought, feel free to ignore this if it's so absurd.

**Reply by tjluoma** [reply]
Although this is veering off-topic, I have done a version of this for myself. I have a list of apps that I have purchased from the Mac App Store (https://mas.luo.ma) which includes the ID number. I have a script that will take a 'friendly' name, and search against that list. As long as it only finds one match, it will take the ID number, and run `mas`.

Of course, trying to do this for the entire Mac App Store would be a huge undertaking which I would not recommend (and `mas` already `mas search {name}` which will let you search the Mac App Store by name, and will return a list of apps, and their ID numbers). But it works well for "These are the apps that I mean when I use these {name}s."

Unfortunately, to go back to the original topic here, the items on that list are only Mac apps, not iOS apps.

**Reply by SMillerDev** [reply]
> and mas already mas search {name} which will let you search the Mac App Store by name, and will return a list of apps, and their ID numbers

There's also `mas lucky {term}` iirc, which will install the first match.

**Comment by SMillerDev** [comment]
Since they are only installable through the app store (as far as I know) you can use `mas`. Unfortunately there is no other way.
